presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoResultSet.java

```diff
@@ -150,7 +153,8 @@ public String getQueryId()
 
     public QueryStats getStats()
     {
-        return QueryStats.create(client.getStats());
+        QueryResults current = client.isValid() ? client.current() : client.finalResults();
```

Comment:
Why? `StatementClient.getStats()` should always work. And if it doesn't, it should be fixed there, not here.